### ISSUE TITLE
Will the DiffRO model be open-sourced?

### ISSUE THREAD

**Issue by Rodriguez009** [FunAudioLLM/CosyVoice#1442]
the DiffRO method mentioned in the paper, seems interesting and very useful! 
wonder if the reward models can be released, so we can Optimize our own LLM

**Comment by aluminumbox** [comment]
you can implement it according to the paper, for example asr reward, there is no plan to add it in our repo yet

**Comment by Ferdydh** [comment]
@Rodriguez009 

i am implementing my own diffro (with minor modifications), so stay tuned :)

**Comment by JohnHerry** [comment]
> [@Rodriguez009](https://github.com/Rodriguez009)
> 
> i am implementing my own diffro (with minor modifications), so stay tuned :)
where  it is goes now?

**Comment by mukherjeesougata-eros** [comment]
@Ferdydh have you finished implementing DiffRO?

**Comment by Ferdydh** [comment]
hi sorry for the late reply. i implemented it and the results were terrible, and i think the problem lied on how diffro fundamentally works, in that:
- the reward model could be unreliable, thus providing bad gradients directly to the model
- llm might experience catastrophic forgetting
- training becomes much slower

for this reason, I did not intend to publish&clean my code, but I made my repo public for those who are interested
https://github.com/Ferdydh/rewards-tts 
@mukherjeesougata-eros 
@JohnHerry

**Comment by JohnHerry** [comment]
Thank you very much for the feedback, I think the main factor about RL is the quality of reward model and training dataset. so it is do really hard for independent developers or `small factories` ‌(>_<)‌
